lib/services/metadata_provider.dart

```diff
@@ -55,23 +53,52 @@ final AutoDisposeFutureProviderFamily<MetadataProvider?, MetadataRequest>
 
   metadataProviderLogger.fine("Fetching metadata for '${request.item.name}' (${request.item.id})");
 
-  //TODO fetch from downloads once implemented
-  
-  final itemInfo = await jellyfinApiHelper.getItemById(request.item.id);
+  BaseItemDto? itemInfo;
+  try {
+    final downloadItem = await downloadsService.getSongInfo(id: request.item.id);
+    if (downloadItem != null) {
+      metadataProviderLogger.fine("Got offline metadata for '${request.item.name}'");
+      itemInfo = downloadItem.baseItem!;
+    }
+  } catch (e) {
+    metadataProviderLogger.warning("Couldn't get the offline metadata for track '${request.item.name}'");
+  }
 
+  // try to fetch from server if not found in downloads
+  itemInfo ??= await jellyfinApiHelper.getItemById(request.item.id);
```

Comment:
Because that one doesn't contain the needed fields (mediaStreams). I guess I could put a check before that...

Reply:
Ah, makes sense.  A check if the provided BaseItemDto is already sufficient would indeed be nice.  It would also be a bit more consistent with other code if we always fetched the metadata from the server in online mode instead of using the downloaded set.